### PR TITLE
secondlife/viewer#1418 Mesh Upload - Physics - Bounding Box is not working as expected

### PR DESCRIPTION
**Pull request by LLGuru**
Before this change:
image

After this change:
image

### CODE REVIEW

File: indra/newview/llmodelpreview.cpp [hunk 1]
```diff
@@ -1067,6 +1069,29 @@ void LLModelPreview::loadModelCallback(S32 loaded_lod)
     { //only replace given LoD
         mModel[loaded_lod] = mModelLoader->mModelList;
         mScene[loaded_lod] = mModelLoader->mScene;
+
+        // Duplicate the model if it is an internal bounding box model
+        if (loaded_lod == LLModel::LOD_PHYSICS &&
+            mBaseModel.size() > 1 && // This makes sense for multiple models only
+            mModelLoader->mModelList.size() == 1 && // Just on the off-chance
+            mModelLoader->mScene.size() == 1 &&     // Just on the off-chance
+            std::filesystem::path(mModelLoader->mFilename).filename() == "cube.dae")
```
Comment (marchcat) [inline indra/newview/llmodelpreview.cpp:1078]: Hardcoded filename?

File: indra/newview/llmodelpreview.cpp [hunk 2]
```diff
@@ -1163,6 +1188,17 @@ void LLModelPreview::loadModelCallback(S32 loaded_lod)
                                     LLFloaterModelPreview::addStringToLog(out, false);
                                 }
                                 mModel[loaded_lod][idx]->mLabel = name;
+                                // Rename the correspondent instance as well
+                                [&]()
```
Comment (marchcat) [inline indra/newview/llmodelpreview.cpp:1192]: Why?

Comment (LLGuru) [reply indra/newview/llmodelpreview.cpp:1192]: > Why?

Just to quick exit from two-level loop

File: indra/newview/llmodelpreview.cpp (hunk 1 above)
Comment (marchcat) [reply indra/newview/llmodelpreview.cpp:1078]: Ah, ok. Thanks.

**Review by marchcat — APPROVED** [review]
Please wait for the Maint X merge before rebasing this.

**Pull request merged**